blog/views.py

```diff
@@ -15,27 +15,38 @@ def homepage(request):
 
 def post_details(request, slug):
     post = Post.objects.filter(slug=slug)
-    p = get_object_or_404(Post, slug=slug)
+    get_post = get_object_or_404(Post, slug=slug)
 
     if post.exists():
         post = post[0]
     else:
         return HttpResponse("Page not found")
 
     comments = post.comments.filter(approve=True)
-    comment_author = request.user
+    user = request.user
     new_comment = None
+    comment_form = AddCommentForm()
 
     if request.method == "POST":
+        if request.user.is_authenticated:
+            user = request.user
+
+            # checking does user liked actual post or not
+            if get_post.like.filter(id=user.id).exists():
+                get_post.like.remove(user.id)
+            else:
+                get_post.like.add(user.id)
```

Comment:
When user gets to the details of the post, function checks in database that does if the user already liked that post or not.

If not, the user can leave a like under the post, similarly when user can undo the action.